packages/dd-trace/src/appsec/sdk/utils.js

```diff
@@ -1,8 +1,27 @@
 'use strict'
 
 function getRootSpan (tracer) {
-  const span = tracer.scope().active()
-  return span && span.context()._trace.started[0]
+  let span = tracer.scope().active()
+  if (!span) return
+
+  const context = span.context()
+  const started = context._trace.started
+
+  let parentId = context._parentId
+  while (parentId) {
+    const parent = started.find(s => s.context()._spanId === parentId)
+    const pContext = parent?.context()
+
+    if (pContext) {
```

Comment:
If `pContext` is null this will lead to an infinite loop, I suggest to check it first:
```
if (!pContext) {
   break;
}
```

Reply:
fixed and added a new test case for it